src/Containers/NoteForm/NoteForm.js

```diff
@@ -1,47 +1,136 @@
 import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
 import { connect } from 'react-redux';
 import { setError } from '../../actions';
+import API from '../../utils/api';
 
 class NoteForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      ...this.props.note
+      showPopUp: true,
+      id: this.props.id,
+      title: this.props.title,
+      issues: this.props.issues
     }
   }
 
-  handleChange = (e) => {
-    const index = this.state.issues.findIndex((issue) => issue.id == e.target.id)
-    console.log('that', index)
-    const issues = this.state.issues.slice()
-    issues[index].body = e.target.value
-    this.setState({ issues })
+  getIndex = (id) => {
+    return this.state.issues.findIndex((issue) => issue.id == id);
+  }
+
+  createIssuesCopy = () => {
+    return this.state.issues.slice();
+  }
+
+  handleTitleChange = (e) => {
+    this.setState({ title: e.target.value });
   } 
 
-  render() {
-    const { title, issues } = this.state
-    return (
-      <div className='overlay-div'>
-        <form className='note-pop-up'>
-          <h3>{title}</h3>
-          <ul>
-            {
-              issues.filter(issue => !issue.completed).map(issue => <li><span><i class="fas fa-square"></i> <input id={issue.id} onChange={this.handleChange} value={issue.body}></input></span> <i class="fas fa-times"></i></li>)
-            }
-          </ul>
-          <i className="fas fa-plus-circle form-add-icon"></i>
-          <h4>Completed</h4>
-          <ul>
-            {
-              issues.filter(issue => issue.completed).map(issue => <li><span><i class="fas fa-check-square"></i> <span contentEditable={true}>{issue.body}</span></span> <i class="fas fa-times"></i></li>)
-            }
-          </ul>
-          <i class="fas fa-save"></i>
-          <i class="fas fa-trash-alt"></i>
-        </form>
-      </div>
+  setIssuesInState = (issues) => {
+    this.setState({ issues });
+  }
+
+  closePopUp = () => {
+    this.formRef.reset();
```

Comment:
Does formRef just reset the form? Was there a reason to do that vs. using setState and setting them to " "? We can also discuss this when we meet up, it might make it easier to explain 😄